### ISSUE TITLE
--data-dir=~/something will not work for validators

### ISSUE THREAD

**Issue by mratsim** [status-im/nimbus-eth2#2164]
When passing `--data-dir=~/something`, the correct path will be found for slashing protection, the node DB but local validators won't be attached.

Passing `--data-dir="$HOME"/something` works instead as it's expanded by the shell I suppose.

**Comment by arnetheduck** [comment]
https://unix.stackexchange.com/questions/151850/why-doesnt-the-tilde-expand-inside-double-quotes - tilde is a bash thing (as is `$HOME`, but with different expansion rules) - I'm not sure what the ticket is about though, ie nbc can't do anything about this..

**Comment by mratsim** [comment]
I didn't use double-quote though.
I think confutils should handle `~` in CLI as ~ is pretty common

**Comment by arnetheduck** [comment]
well, that's a bit weird - why should it handle one platform specific expansion like `~` but not others like `$`? these kinds of well-meaning "smart" features tend to lead to lead to even more confusion because now there are multiple places where it _may_ have been expanded, and you have to make sure to implement exactly the same expansion rules as posix etc

**Comment by arnetheduck** [comment]
there's also the cruft argument - ie you like `~`, another one likes `$`, a third uses some different shell with different expansion rules.. should we support all weird rules that "someone" likes? (ie I know `~` exists but never use it)

**Comment by mratsim** [comment]
There is no need to support $ because the shell always expand it. But `~` might have conflict because confutils is using `=` instead of spaces to separate a command from its argument.

It's not about me it's about either supporting `~` fully or rejecting it `~` fully, but not the current half-way where you have your data put in the proper place but your validators are not loaded.

**Comment by arnetheduck** [comment]
the point is more that you made a spelling mistake: your shell should have expanded it but didn't, according to its rules - what other spelling mistakes should we compensate for, and why is `~` special?

**Comment by mratsim** [comment]
What spelling mistake are you talking about?

**Comment by arnetheduck** [comment]
the mistake of using `~` in a way that's not supported by bash, and thus expecting it it work

**Comment by stefantalpalaru** [comment]
> using `~` in a way that's not supported by bash

That's correct. Bash will not do tilde expansion in a string of the form `--foo=~/bar` due to its position. It would have done it if we were using proper POSIX conventions: `--foo ~/bar`.

https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html

> we should have a proper error message or this will cause debugging headaches for end users

I agree. Should be trivial to check for a tilde at the beginning of strings that are supposed to be used as paths.

**Comment by mratsim** [comment]
If chronicles is not following convention, it's our problem to solve we should not dump that to the end users.
